store/sqlstore/file_info_store.go

```diff
@@ -95,6 +96,72 @@ func (fs SqlFileInfoStore) Get(id string) (*model.FileInfo, *model.AppError) {
 	return info, nil
 }
 
+func (fs SqlFileInfoStore) GetWithOptions(opt *model.GetFilesOptions) ([]*model.FileInfo, *model.AppError) {
+	if opt == nil {
+		opt = &model.GetFilesOptions{}
+	}
+
+	query := fs.getQueryBuilder().
+		Select("FileInfo.*").
+		From("FileInfo").
+		LeftJoin("Posts ON FileInfo.PostId = Posts.Id") // Left join to get files that aren't attached to a post
+
+	if len(opt.ChannelIds) > 0 {
+		query = query.Where(sq.Eq{"Posts.ChannelId": opt.ChannelIds})
+	}
+
+	if len(opt.UserIds) > 0 {
+		query = query.Where(sq.Eq{"FileInfo.CreatorId": opt.UserIds})
+	}
+
+	if opt.Since > 0 {
+		query = query.Where(sq.Gt{"FileInfo.CreateAt": opt.Since})
+	}
+
+	if !opt.IncludeDeleted {
+		query = query.Where("FileInfo.DeleteAt = 0")
+	}
+
+	sortDirection := model.FILE_SORT_ORDER_ASCENDING
+	if opt.SortDirection != "" {
+		sortDirection = opt.SortDirection
+	}
+	if opt.SortBy != "" {
+		switch opt.SortBy {
+		case model.FILE_SORT_BY_CREATED, model.FILE_SORT_BY_SIZE:
+			query = query.OrderBy(fmt.Sprintf("FileInfo.%s %s", opt.SortBy, sortDirection))
+		case model.FILE_SORT_BY_USERNAME:
```

Comment:
I'm struggling to think of a case where this sort would be useful, and the join cost is non-trivial here. I'd suggest skipping this one.

Reply:
I wasn't sure of this either.  

Since the caller is able to filter by user id and channel id, I thought the caller might want to sort by the names as well (because sorting by id didn't seem useful).

I'll defer to your call.

Reply:
I propose we remove this for now -- easy to revisit if we can come up with a compelling scenario, but simpler code is always easier to maintain and test.